spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/StatsdReporter.java

```diff
@@ -19,154 +34,112 @@
-        }
-    }
-
     @Override
-    public void report(SortedMap<String, Gauge> gauges, SortedMap<String, Counter> counters, SortedMap<String, Histogram> histograms, SortedMap<String, Meter> meters, SortedMap<String, Timer> timers) {
+    public void report(SortedMap<String, Gauge> gauges, SortedMap<String, Counter> counters,
+            SortedMap<String, Histogram> histograms, SortedMap<String, Meter> meters, SortedMap<String, Timer> timers) {
         try (DatagramSocket socket = new DatagramSocket()) {
-            try {
-                reportGauges(gauges, socket);
-                reportCounters(counters, socket);
-                reportHistograms(histograms, socket);
-                reportMeters(meters, socket);
-                reportTimers(timers, socket);
-            } catch (StatsdReporterException e) {
-                logger.warn("Unable to send packets to StatsD", e);
-            }
+            reportGauges(getFilteredMetrics(gauges), socket);
+            reportCounters(getFilteredMetrics(counters), socket);
+            reportHistograms(getFilteredMetrics(histograms), socket);
+            reportMeters(getFilteredMetrics(meters), socket);
+            reportTimers(getFilteredMetrics(timers), socket);
+        } catch (StatsdReporterException e) {
+            logger.warn("Unable to send packets to StatsD", e);
         } catch (IOException e) {
             logger.warn("StatsD datagram socket construction failed", e);
         }
     }
 
+    private <T extends Metric> SortedMap<String, T> getFilteredMetrics(SortedMap<String, T> metrics) {
+        final TreeMap<String, T> filteredMetrics = new TreeMap<>();
+        for (Map.Entry<String, T> entry : metrics.entrySet()) {
+            if (filter.matches(entry.getKey(), entry.getValue())) {
+                filteredMetrics.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return Collections.unmodifiableSortedMap(filteredMetrics);
+    }
+
     private void reportGauges(SortedMap<String, Gauge> gauges, DatagramSocket socket) {
-        gauges.forEach((name, value) ->
-                send(socket, metricFormatter.buildMetricString(name, value.getValue(), GAUGE))
-        );
+        gauges.forEach((name, value) -> send(socket, metricFormatter.buildMetricString(name, value.getValue(), GAUGE)));
     }
 
     private void reportCounters(SortedMap<String, Counter> counters, DatagramSocket socket) {
         counters.forEach((name, value) ->
-                // counter is reported as a gauge because StatsD treats new values as increments
-                // but not as a final value e.g. sending 'foo:1|c' to StatsD increments 'foo' by 1
-                send(socket, metricFormatter.buildMetricString(name, value.getCount(), GAUGE))
-        );
+        // counter is reported as a gauge because StatsD treats new values as increments
+        // but not as a final value e.g. sending 'foo:1|c' to StatsD increments 'foo' by
+        // 1
```

Comment:
can we move `1` one line above to ease the reading?

Reply:
Moved it.